src/megaclient.cpp

```diff
@@ -34,6 +35,80 @@ namespace mega {
 // FIXME: instead of copying nodes, move if the source is in the rubbish to reduce node creation load on the servers
 // FIXME: prevent synced folder from being moved into another synced folder
 
+
+void MegaClientAsyncQueue::push(std::function<void(SymmCipher&)> f)
+{
+    std::lock_guard<std::mutex> g(mMutex);
```

Comment:
The lock should only protect access to `mQueue`

Reply:
good point, I think that will be fine, adjusted